File: app/src/main/java/com/digicoffer/lauditor/CommonFiles/GlobalFiles/NotificationCountApi.java
package com.digicoffer.lauditor.CommonFiles.GlobalFiles;

import android.content.Context;
import android.util.Log;
import android.view.View;

import com.digicoffer.lauditor.Dashboard.DahboardModels.NotificationCountModel;
import com.digicoffer.lauditor.Webservice.AsyncTaskCompleteListener;
import com.digicoffer.lauditor.Webservice.HttpResultDo;
import com.digicoffer.lauditor.Webservice.CommonApiHelper.WebServiceHelper;

import org.json.JSONArray;
import org.json.JSONObject;

import java.util.ArrayList;

public class NotificationCountApi implements AsyncTaskCompleteListener {

    Context context;
    public static ArrayList<NotificationCountModel> notificationList = new ArrayList<>();
    public static int unreadNotificationCount = 0;

    public NotificationCountApi(Context context) {
        this.context = context;
    }

    @Override
    public void onClick(View v) { }

    @Override
    public void onAsyncTaskComplete(HttpResultDo httpResult) {
        if (httpResult.getResult() == WebServiceHelper.ServiceCallStatus.Success) {
            try {
                if (httpResult.getRequestType().equalsIgnoreCase("GET NOTIFICATION COUNT")) {

                    JSONObject response = new JSONObject(httpResult.getResponseContent());
                    JSONObject dataObj = response.getJSONObject("data");
                    JSONArray notifications = dataObj.getJSONArray("notifications");

                    notificationList.clear();
                    unreadNotificationCount = 0;

                    for (int i = 0; i < notifications.length(); i++) {

                        JSONObject obj = notifications.getJSONObject(i);

                        NotificationCountModel model = new NotificationCountModel();
                        model.setId(obj.getString("id"));
                        model.setMessage(obj.getString("message"));
                        model.setTimestamp(obj.getString("timestamp"));
                        model.setStatus(obj.getString("status"));

                        notificationList.add(model);

                        // 🔥 Count only unread
                        if (model.getStatus().equalsIgnoreCase("unread")) {
                            unreadNotificationCount++;
                        }
                    }

                    Log.d("NOTIFY_API", "Unread Count: " + unreadNotificationCount);

                    // update UI badge
                    if (Constants.notifyBadge != null) {
                        if (unreadNotificationCount > 0) {
                            Constants.notifyBadge.setText(String.valueOf(unreadNotificationCount));
                        } else {
                            Constants.notifyBadge.setVisibility(View.GONE);
                        }
                    }
                }

            } catch (Exception e) {
                e.printStackTrace();
            }
        }
    }

    // 🔥 Call This to Fetch Notification List
    public void fetchNotificationCount() {
        WebServiceHelper.callHttpWebService(
                this,
                context,
                WebServiceHelper.RestMethodType.GET,
                "notification",
                "GET NOTIFICATION COUNT",
                ""
        );
    }
}

File: app/src/main/java/com/digicoffer/lauditor/CommonFiles/GlobalFiles/NotificationCountApi.kt
package com.digicoffer.lauditor.CommonFiles.GlobalFiles

import android.content.Context
import android.util.Log
import android.view.View
import com.digicoffer.lauditor.Dashboard.DahboardModels.NotificationCountModel
import com.digicoffer.lauditor.Webservice.AsyncTaskCompleteListener
import com.digicoffer.lauditor.Webservice.CommonApiHelper.WebServiceHelper
import com.digicoffer.lauditor.Webservice.HttpResultDo
import org.json.JSONArray
import org.json.JSONObject
import java.util.ArrayList

class NotificationCountApi(private val context: Context) : AsyncTaskCompleteListener {

    override fun onClick(v: View) {}

    override fun onAsyncTaskComplete(httpResult: HttpResultDo) {
        if (httpResult.result == WebServiceHelper.ServiceCallStatus.Success) {
            try {
                if (httpResult.requestType.equals("GET NOTIFICATION COUNT", ignoreCase = true)) {
                    val response = JSONObject(httpResult.responseContent ?: "")
                    val dataObj = response.getJSONObject("data")
                    val notifications = dataObj.getJSONArray("notifications")

                    notificationList.clear()
                    unreadNotificationCount = 0

                    for (i in 0 until notifications.length()) {
                        val obj = notifications.getJSONObject(i)

                        val model = NotificationCountModel()
                        model.id = obj.getString("id")
                        model.message = obj.getString("message")
                        model.timestamp = obj.getString("timestamp")
                        model.status = obj.getString("status")

                        notificationList.add(model)

                        if (model.status.equals("unread", ignoreCase = true)) {
                            unreadNotificationCount++
                        }
                    }

                    Log.d("NOTIFY_API", "Unread Count: $unreadNotificationCount")

                    val badge = Constants.notifyBadge
                    if (badge != null) {
                        if (unreadNotificationCount > 0) {
                            badge.text = unreadNotificationCount.toString()
                        } else {
                            badge.visibility = View.GONE
                        }
                    }
                }
            } catch (e: Exception) {
                e.printStackTrace()
            }
        }
    }

    fun fetchNotificationCount() {
        WebServiceHelper.callHttpWebService(
            this,
            context,
            WebServiceHelper.RestMethodType.GET,
            "notification",
            "GET NOTIFICATION COUNT",
            ""
        )
    }

    companion object {
        @JvmField
        val notificationList = ArrayList<NotificationCountModel>()

        @JvmField
        var unreadNotificationCount = 0
    }
}
